lib/facter/resolvers/freebsd/system_memory_resolver.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Freebsd
+      class SystemMemory < BaseResolver
+        @semaphore = Mutex.new
+        @fact_list ||= {}
+        class << self
+          private
+
+          def post_resolve(fact_name)
+            @fact_list.fetch(fact_name) { calculate_system_memory(fact_name) }
+          end
+
+          def calculate_system_memory(fact_name)
+            read_total_memory_in_bytes
+            read_available_memory_in_bytes
+
+            @fact_list[:used_bytes] = @fact_list[:total_bytes] - @fact_list[:available_bytes]
+            @fact_list[:capacity] = compute_capacity(@fact_list[:used_bytes], @fact_list[:total_bytes])
+
+            @fact_list[fact_name]
+          end
+
+          def read_available_memory_in_bytes
+            output = Facter::Core::Execution.execute('vmstat -H', logger: log)
```

Comment:
briefly wondered if it'd make sense to use `--libxo`

Reply:
Makes sense, yes!  Let's do this :wink:

Reply:
what hath i wrought?? 😅

Reply:
btw, this means we're dropping support for FreeBSD 10.4… https://www.freebsd.org/cgi/man.cgi?query=vmstat&apropos=0&sektion=0&manpath=FreeBSD+10.4-RELEASE&arch=default&format=html

then again, that's no longer supported, so… https://www.freebsd.org/security/security.html#sup